### PR TITLE
Release 0.7.3

### PR DESCRIPTION
**Pull request by huard**
## Overview

- update CHANGES with 0.7.3 release notes.
- Pin xesmf >= 0.5.3.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@ cftime
 pywps==4.4
 xclim==0.24.0
 xarray
+xesmf>=0.5.3
```
Comment (tlvu) [inline requirements.txt:5]: Houston, on Pypi https://pypi.org/project/xesmf/#history xesmf is only 0.3.0 !

Comment (huard) [reply requirements.txt:5]: Yup. We're using https://github.com/pangeo-data/xESMF

Comment (huard) [reply requirements.txt:5]: The core developer vanished, so we forked the repo and maintain it elsewhere.

Comment (tlvu) [reply requirements.txt:5]: My point is that line will never work if xesmf on Pypi is stuck at 0.3.0.

Comment (huard) [reply requirements.txt:5]: To be clear, xESMF cannot be installed with pip. It has to be installed with conda. 
However, once it's installed, my understanding is that we can still ask setup.py to check that the installed version meets some requirements, irrespective of how it was installed.

**Comment by tlvu** [comment]
Testing building Finch docker from this branch with xESMF 0.5.3 conda package.

File: environment-docs.yml [hunk 2]
```diff
@@ -10,3 +10,4 @@ dependencies:
 - ipython
 - xclim==0.24
 - matplotlib
+- xesmf>=0.5.3
```
Comment (tlvu) [inline environment-docs.yml:13]: environment-docs.yml is only to build docs on ReadTheDocs.  Do we need xesmf to build docs (it wasn't there before)?  In general we want to keep this env file to the minimum to not burst the memory limit on ReadTheDocs.

Comment (aulemahal) [reply environment-docs.yml:13]: I see you're right! I didn't realize we didn't install finch on RTD.

Comment (aulemahal) [inline environment-docs.yml:13]:
```suggestion
```

**Comment by huard** [comment]
@aulemahal ok to merge ? notebooks pass on my end.

**Review by aulemahal — APPROVED** [review]

**Comment by huard** [comment]
Argh ! J'ai oublié bumpversion ! Je le fais sur master direct...

**Pull request merged**